blockscout_mcp_server/tools/block_tools.py

```diff
@@ -1,32 +1,78 @@
-from typing import Annotated, Dict
+from typing import Annotated, Dict, Optional
+import asyncio
+import json
 from pydantic import Field
 from blockscout_mcp_server.tools.common import make_blockscout_request, get_blockscout_base_url
 from mcp.server.fastmcp import Context
 
 async def get_block_info(
     chain_id: Annotated[str, Field(description="The ID of the blockchain")],
     number_or_hash: Annotated[str, Field(description="Block number or hash")],
-    ctx: Context
-) -> Dict:
+    ctx: Context,
+    include_transactions: Annotated[Optional[bool], Field(description="If true, includes a list of transaction hashes from the block. Use this option sparingly as it requires an extra API call.")] = False
+) -> str:
     """
     Get block information like timestamp, gas used, burnt fees, transaction count etc.
+    Can optionally include the list of transaction hashes contained in the block. This must be requested only when getting block transactions is really needed.
```

Comment:
```suggestion
    Can optionally include the list of transaction hashes contained in the block. Transaction hashes are omitted by default; request them only when you truly need them, because on high-traffic chains the list may exhaust the context.
```